modules/starboard.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+# Starboard commands
+module Starboard
+  extend Discordrb::Commands::CommandContainer
+end
+
+# Reaction event handler
+QBot.bot.reaction_add do |event|
+  scfg = ServerConfig[event.server.id]
+  break if scfg.modules_conf['disabled'].include? 'starboard'
+
+  # Define functions for embed
+  ## Get author
+  def embed_author(user)
+    {
+      icon_url: user.avatar_url,
+      name: user.username
+    }
+  end
+
+  ## Get message attachment
+  def embed_image(message)
+    return unless message.attachments.first
+
+    {
+      url: message.attachments.first.url
+    }
+  end
+
+  ## Get field text
+  def embed_fields(event)
+    [{
+      name: 'Message',
+      value: format('[#%<name>s](%<link>s)', name: event.channel.name, link: event.message.link)
+    }]
+  end
+
+  # Get variables from db
+  emoji_name = (scfg.options['starboard-emoji'] || QBot.config.global.starboard.emoji)
```

Comment:
pull these lines out into another method and use destructuring assignment to get the values in